### PR TITLE
Refactor `restCallVM` to `restCallC`

### PR DESCRIPTION
**Pull request by EnoF**
VM -> Viewing resources
C -> Creating or Updating resources
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/EnoF/con-rest/pull/37%23issuecomment-69165639%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23issuecomment-69165688%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22641586%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22641637%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22641646%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22643963%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22643985%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22643998%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22646091%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22646092%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22646093%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/37%23issuecomment-69165639%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Looks%20good%20to%20me%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-01-08T11%3A08%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-01-08T11%3A08%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20866e86756f6949ac88e67f0b89dd44806033521c%20app/dao/requestDAO.js%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22641586%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20are%20you%20using%20the%20requestRaw%20method%20here%3F%22%2C%20%22created_at%22%3A%20%222015-01-08T09%3A18%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%2C%20%7B%22body%22%3A%20%22Reason%20being%2C%20for%20the%20execution%20we%20need%20the%20RAW%20response%20to%20display%20it%20to%20the%20user%20%28including%20the%20status%20code%20etc%29%22%2C%20%22created_at%22%3A%20%222015-01-08T10%3A17%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-08T11%3A09%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/dao/requestDAO.js%3AL1-35%22%7D%2C%20%22Pull%20866e86756f6949ac88e67f0b89dd44806033521c%20app/widgets/callRegistrator/callRegistrator.js%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22641637%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22do%20we%20need%20an%20empty%20link%20function%3F%22%2C%20%22created_at%22%3A%20%222015-01-08T09%3A19%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%2C%20%7B%22body%22%3A%20%22Nope%20we%20don%27t%2C%20this%20was%20a%20left%20over%20from%20some%20refactoring%20testing%2C%20will%20remove%20it.%20Thnx.%22%2C%20%22created_at%22%3A%20%222015-01-08T10%3A17%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-08T11%3A09%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/widgets/callRegistrator/callRegistrator.js%3AL10-25%22%7D%2C%20%22Pull%20866e86756f6949ac88e67f0b89dd44806033521c%20test/unit/app/conRestTestApp.js%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/37%23discussion_r22641646%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22badadu%20banana%3F%22%2C%20%22created_at%22%3A%20%222015-01-08T09%3A19%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Abanana%3A%20%22%2C%20%22created_at%22%3A%20%222015-01-08T10%3A18%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-08T11%3A09%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TheBay0r%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/unit/app/conRestTestApp.js%3AL23-35%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/TheBay0r%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8627051%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/TheBay0r'></a>

- [x] <a href='#crh-comment-Pull 866e86756f6949ac88e67f0b89dd44806033521c app/dao/requestDAO.js 15'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/37#discussion_r22641586'>File: app/dao/requestDAO.js:L1-35</a></b>
- <a href='https://github.com/dominikkukacka'></a> Why are you using the requestRaw method here?
- <a href='https://github.com/EnoF'></a> Reason being, for the execution we need the RAW response to display it to the user (including the status code etc)
- [x] <a href='#crh-comment-Pull 866e86756f6949ac88e67f0b89dd44806033521c app/widgets/callRegistrator/callRegistrator.js 12'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/37#discussion_r22641637'>File: app/widgets/callRegistrator/callRegistrator.js:L10-25</a></b>
- <a href='https://github.com/dominikkukacka'></a> do we need an empty link function?
- <a href='https://github.com/EnoF'></a> Nope we don't, this was a left over from some refactoring testing, will remove it. Thnx.
- [x] <a href='#crh-comment-Pull 866e86756f6949ac88e67f0b89dd44806033521c test/unit/app/conRestTestApp.js 7'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/37#discussion_r22641646'>File: test/unit/app/conRestTestApp.js:L23-35</a></b>
- <a href='https://github.com/dominikkukacka'></a> badadu banana?
- <a href='https://github.com/EnoF'></a> :banana:
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/37#issuecomment-69165639'>General Comment</a></b>
- <a href='https://github.com/TheBay0r'></a> Looks good to me :+1:

<a href='https://www.codereviewhub.com/EnoF/con-rest/pull/37?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/EnoF/con-rest/pull/37?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/EnoF/con-rest/pull/37?approve=0'></a>&nbsp;<a href='https://github.com/EnoF/con-rest/pull/37'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by EnoF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/EnoF/con-rest/pull/37'>Refresh</a>.<a href='#crh-update-none'></a>

File: app/dao/requestDAO.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    app.factory('requestDAO', function requestDAOScope($http, $q, DAO) {
+        function requestDAO() {
+            this.extend = 'DAO';
+
+            this.public = {
+                getCall: function getCall(id) {
+                    return this.private.request('GET', '/api/requests/' + id);
+                },
+                executeCall: function executeCall(id) {
+                    return this.private.requestRaw('POST', '/api/requests/' + id +
```
Comment (dominikkukacka) [inline app/dao/requestDAO.js]: Why are you using the requestRaw method here?

File: app/widgets/callRegistrator/callRegistrator.js [hunk 2]
```diff
@@ -10,12 +10,15 @@
...
         return {
-            controller: 'restCallVM',
+            controller: 'restCallC',
             restrict: 'EA',
             scope: {
                 request: '=?restCall'
             },
-            templateUrl: 'callRegistrator'
+            templateUrl: 'callRegistrator',
+            link: function callRegistratorConstructor() {
```
Comment (dominikkukacka) [inline app/widgets/callRegistrator/callRegistrator.js]: do we need an empty link function?

File: test/unit/app/conRestTestApp.js [hunk 3]
```diff
@@ -23,8 +23,12 @@
                 name: 'fakeCall',
                 url: 'https://fake.url',
                 method: 'PUT',
-                data: { ba: 'nana' },
-                headers: { to: 'ken' }
+                data: {
+                    ba: 'nana'
```
Comment (dominikkukacka) [inline test/unit/app/conRestTestApp.js]: badadu banana?

File: app/dao/requestDAO.js (hunk 1 above)
Comment (EnoF) [inline app/dao/requestDAO.js]: Reason being, for the execution we need the RAW response to display it to the user (including the status code etc)

File: app/widgets/callRegistrator/callRegistrator.js (hunk 2 above)
Comment (EnoF) [inline app/widgets/callRegistrator/callRegistrator.js]: Nope we don't, this was a left over from some refactoring testing, will remove it. Thnx.

File: test/unit/app/conRestTestApp.js (hunk 3 above)
Comment (EnoF) [inline test/unit/app/conRestTestApp.js]: :banana:

**Comment by TheBay0r** [comment]
Looks good to me :+1:

**Comment by TheBay0r** [comment]
:shipit:

File: app/dao/requestDAO.js (hunk 1 above)
Comment (TheBay0r) [inline app/dao/requestDAO.js]: :+1:<a href='#crh-update-none'></a>

File: app/widgets/callRegistrator/callRegistrator.js (hunk 2 above)
Comment (TheBay0r) [inline app/widgets/callRegistrator/callRegistrator.js]: :+1:<a href='#crh-update-none'></a>

File: test/unit/app/conRestTestApp.js (hunk 3 above)
Comment (TheBay0r) [inline test/unit/app/conRestTestApp.js]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**